plugins/permission-node/src/ServerPermissionClient.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { TokenManager, ServerTokenManager } from '@backstage/backend-common';
+import { Config } from '@backstage/config';
+import {
+  AuthorizeRequest,
+  AuthorizeRequestOptions,
+  AuthorizeResponse,
+  AuthorizeResult,
+  DiscoveryApi,
+  PermissionClient,
+  PermissionClientInterface,
+} from '@backstage/plugin-permission-common';
+
+/**
+ * A thin wrapper around
+ * {@link @backstage/plugin-permission-common#PermissionClient} that allows all
+ * backend-to-backend requests.
+ * @public
+ */
+export class ServerPermissionClient implements PermissionClientInterface {
```

Comment:
I was hoping we could discuss the name of this further, glad you pointed it out. We currently have a frontend `[PermissionApi](https://github.com/backstage/backstage/blob/393f107893ec4ab4903d9022b8c3d0b6d0c4970d/plugins/permission-react/src/apis/PermissionApi.ts#L29)` - do you think it would be okay to name this the same thing? Might cause accidental imports of the wrong one for IDEs. Honestly I think the best name for the interface might be just `PermissionClient` and then the existing PermissionClient implementation would have to become `DefaultPermissionClient` or `CommonPermissionClient` or something like that?

If just calling it `PermissionApi` follow existing patterns though, I'm happy to go that route.

Reply:
Duplicate comment by me here 😅 , but that's good too I guess :p

I think best avoid `PermissionApi`, definitely not worth the frontend vs backend discovery nightmare. Thinking forward a bit I do think we should find some suffix that fits well for the backend, could use one of the ones we didn't use in the frontend, for example `*Service`. Either way we're gonna have to bikeshed that plenty as we take a look at the backend system next year, for now I'd just pick a relatively neutral name, maybe one that would fit with a suffix later

Reply:
We decided to go with `PermissionAuthorizer` - see [the newest commit here](https://github.com/backstage/backstage/pull/8357/commits/3a061cedd77b6786d0057c6410376fddb8e8c34e).